modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/FieldBase/ReactFieldBase.es.js

```diff
@@ -77,131 +82,114 @@ function FieldBase({
+								<ClayIcon symbol="hr" />
 							</ClayButton>
-						</div>
-					)}
-
-					{((label && showLabel) ||
-						required ||
-						tooltip ||
-						repeatable) && (
-						<p
-							className={classNames({
-								'ddm-empty': !showLabel && !required,
-								'ddm-label': showLabel,
-							})}
+						)}
+
+						<ClayButton
+							className="ddm-form-field-repeatable-add-button p-0"
+							disabled={readOnly}
+							onClick={() =>
+								dispatch({
+									payload: name,
+									type: PAGE_TYPES.REPEATED,
+								})
+							}
+							small
+							title={Liferay.Language.get('duplicate')}
+							type="button"
 						>
-							{label && showLabel && label}
-
-							{required && spritemap && (
-								<span className="reference-mark">
-									<ClayIcon
-										spritemap={spritemap}
-										symbol="asterisk"
-									/>
-								</span>
-							)}
-
-							{tooltip && (
-								<div className="ddm-tooltip">
-									<ClayIcon
-										data-tooltip-align="right"
-										spritemap={spritemap}
-										symbol="question-circle-full"
-										title={tooltip}
-									/>
-								</div>
-							)}
-						</p>
-					)}
-
-					{children}
-
-					{nestedFields && (
-						<div className="ddm-field-types-base__nested">
-							<PageRendererAdapter
-								context={context}
-								editable={editable}
-								rows={getDefaultRows(nestedFields)}
-								spritemap={spritemap}
-							/>
-						</div>
-					)}
-
-					{localizedValueArray.length > 0 &&
-						localizedValueArray.map((language) => (
-							<input
-								key={language.name}
-								name={language.name}
-								type="hidden"
-								value={language.value}
-							/>
-						))}
-
-					{tip && <span className="form-text">{tip}</span>}
-
-					{displayErrors && errorMessage && !valid && (
-						<span className="form-feedback-group">
-							<div className="form-feedback-item">
-								{errorMessage}
-							</div>
-						</span>
-					)}
-				</div>
-			</ClayTooltipProvider>
-		</ClayIconSpriteContext.Provider>
+							<ClayIcon symbol="plus" />
+						</ClayButton>
+					</div>
+				)}
+
+				{((label && showLabel) ||
+					required ||
+					tooltip ||
+					repeatable) && (
+					<p
+						className={classNames({
+							'ddm-empty': !showLabel && !required,
+							'ddm-label': showLabel,
+						})}
+					>
+						{label && showLabel && label}
+
+						{required && (
+							<span className="reference-mark">
+								<ClayIcon symbol="asterisk" />
+							</span>
+						)}
+
+						{tooltip && (
+							<span className="ddm-tooltip">
+								<ClayIcon
+									data-tooltip-align="right"
+									symbol="question-circle-full"
+									title={tooltip}
+								/>
+							</span>
+						)}
+					</p>
+				)}
+
+				{children}
+
+				{nestedFields && (
+					<Layout
+						onBlur={onBlur}
+						onChange={onChange}
+						onFocus={onFocus}
+						rows={getDefaultRows(nestedFields)}
+					/>
+				)}
```

Comment:
I still need to work on this case of `nestedFields` so that all components do not need to pass these properties to ReactFieldBase and it communicates directly with PageRenderer.

Reply:
I just added a new commit to cover this use case.